Add the ability to run in Docker

sk4r opened the pull request:
Lets the user run the server with fewer steps.

The Dockerfile takes into consideration some of the workarounds previously listed in the readme and could be simplified once an official build is available.

~Note: the build provided here [vs_server_linux-arm64_1.19.0-rc.1.tar.gz](https://github.com/anegostudios/VintagestoryServerArm64/blob/main/vs_server_linux-arm64_1.19.0-rc.1.tar.gz) seems to be using NET 8 instead of NET 7, hence the `8.0` image tags.~

Tested:
1. run server on Raspberry Pi 5
2. client using 1.19.0-rc.6 is able to connect
3. server saves data with `/stop`
4. running the server again loads properly

See the updated readme for details.

sk4r committed:
Add the ability to run in Docker

Th3Dilli commented:
hey thank you for the contribution


I did not notice that those was net8 so now I have updated it to be net7 and added 1.19.rc.6

If you can update the docker for that, we can then merge this

The reason why it was/is a more complicated install process is because then I do not have to update for every version since all code changes usually happen in other files than the once in the `server` folder but I will keep at least one recent version with full files.
But those might lag behind the official release hence why the merge of the files with the official package should be preferred if possible

sk4r committed:
update to 1.19.0-rc.6, move back to .net 7 from .net 8

sk4r commented:
> hey thank you for the contribution
> 
> I did not notice that those was net8 so now I have updated it to be net7 and added 1.19.rc.6
> 
> If you can update the docker for that, we can then merge this
> 
> The reason why it was/is a more complicated install process is because then I do not have to update for every version since all code changes usually happen in other files than the once in the `server` folder but I will keep at least one recent version with full files. But those might lag behind the official release hence why the merge of the files with the official package should be preferred if possible

had to tweak a bit for NET 7, but just tested rc6 and it works

sk4r committed:
Merge branch 'anegostudios:main' into docker-compose

sk4r commented on Dockerfile:
```diff
@@ -0,0 +1,41 @@
+    VintagestoryServer.pdb \
+    VintagestoryServer.runtimeconfig.json \
+    Lib
+RUN tar xzf /tmp/vs_server_linux-arm64_${VS_VERSION}.tar.gz
```
this could be changed to `ADD ./server .` to just use `./server/*` files directly and get rid of the extra download above. However, then I get an error on startup:
```
vsserver_1  | Unhandled exception. System.BadImageFormatException: Bad IL format. The format of the file '/opt/vintagestory/VintagestoryServer.dll' is invalid.
vsserver_1  | Aborted (core dumped)
```

Th3Dilli replied:
what does it return when you do file "/opt/vintagestory/VintagestoryServer.dll"

you need to make sure that when using the repo files you replace the VintagestoryServer.dll that is in the official x64 release with the arm one

sk4r replied:
@Th3Dilli , never mind, I hadn't installed `git-lfs` on the Pi :man_facepalming: 

So that works, now it downloads the specified x64 tarball and copies over from `./server` in this repo.

sk4r committed:
use files from the local server dir

sk4r committed:
fix readme commands
